### PR TITLE
Add score

### LINKED ISSUE #96: Implement a basic search scoring algorithm
This search score should be used as the ORDER BY for the search API, and will be an initial version of the package quality score.

### PR DESCRIPTION
**Pull request by finestructure**
Fixes #96

### CODE REVIEW

**Comment by daveverwer** [comment]
I know you didn't mark this for review, but I wanted to just chime in! 😬

**Comment by finestructure** [comment]
Please do :) I want to give it another pass in the morning, because I’m not happy how `updateStatus` is now setting `status`, `processingStage`, and `score`. I’ll probably at least want to rename this to `updatePackage` but I ran out of time to wrap all that up :)

**Comment by finestructure** [comment]
Oh, and I might also just tack on the oder by clause in the search query.

File: Sources/App/Commands/Common.swift [hunk 1]
```diff
@@ -8,9 +8,23 @@ func updateStatus(application: Application,
...
+                guard let pkgId = pkg.id else {
+                    return application.db.eventLoop.future(error:
+                        AppError.genericError(nil, "updateStatus: pkgId nil for package \(pkg.url)")
+                    )
+                }
+                return Package.query(on: application.db)
+                    .with(\.$repositories)
+                    .with(\.$versions)
+                    .filter(\.$id == pkgId)
+                    .first()
```
Comment (finestructure) [inline Sources/App/Commands/Common.swift:20]: Just saw in Discord that there’s a better way to do this:

```
<@!254296890047332352> You can use `job.$parent.load(on:)` if you have an instance of your model
```

**Comment by finestructure** [comment]
Ok, this is good to go now!

File: Sources/App/Core/Score.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            default    :   score += 20
+        }
+
+        // Stars count
+        switch candidate.likeCount {
+            case      ..<25    :  break
+            case    25..<100   :  score += 10
+            case   100..<500   :  score += 20
+            case   500..<5_000 :  score += 30
+            case 5_000..<10_000:  score += 40
```
[the remark this answers is not in the document]
Comment (finestructure) [reply Sources/App/Core/Score.swift:27]: Umm, what did I mess up? This should be exactly the same as the snippet you posted in the issue 😅

File: Sources/App/Core/Score.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            default:              score += 50
+        }
+
```
[the remark this answers is not in the document]
Comment (finestructure) [reply Sources/App/Core/Score.swift:30]: Only that I’m blind 😅

File: Sources/App/Core/Score.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (daveverwer) [reply Sources/App/Core/Score.swift:27]: The numbers in the scoring aren't quite the same, and they were picked quite carefully. In this version, every gradation gives an equal 10 points to the score, but in the original less points were awarded for the higher tiers.

Screenshot 2020-05-25 at 08 59 55@2x

File: Sources/App/Core/Score.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (finestructure) [reply Sources/App/Core/Score.swift:30]: Just to be clear, the intention was to transfer the algo 1:1.

I’ve only arranged the successive ifs that add to the score as they go past thresholds into switch statements, as I found them easier to parse.

File: Sources/App/Core/Score.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (finestructure) [reply Sources/App/Core/Score.swift:27]: Oh god, I really am blind. It’s +5 and +2 towards the tail end.

File: Sources/App/Core/Score.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (daveverwer) [reply Sources/App/Core/Score.swift:30]: 👍

I don't think this score is perfect by any means, but I do think it's a good place to start from, and certainly the diminishing returns for large numbers of stars is something I think is going to work well.

**Comment by finestructure** [comment]
That's good to go now!

**Pull request merged**